org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/HandleCryptosystemTextbook.java

```diff
@@ -0,0 +1,974 @@
+		cstb.txtChosenPlaintexts.setText("");
+	}
+	
+	
+	
+	private String getCurrentSelectedPlaintexts() {
+		String ret = "";
+		Set<String> keys = currentSelectedPlaintexts.keySet();
+		for(String key : keys) {
+			ret += currentSelectedPlaintexts.get(key);
+		}
+		return ret;
+	}
+	
+	
+	
+	private void setColorPlaintext(CryptosystemTextbookComposite cstb, int idxToSet, Color colorBG, Color colorFG) {
+		switch(idxToSet) {
+			case 0:
+				cstb.txtFirstPlaintext.setBackground(colorBG);
+				cstb.txtFirstPlaintext.setForeground(colorFG);
+				break;
+			
+			case 1:
+				cstb.txtSecondPlaintext.setBackground(colorBG);
+				cstb.txtSecondPlaintext.setForeground(colorFG);
+				break;
+			
+			case 2:
+				cstb.txtThirdPlaintext.setBackground(colorBG);
+				cstb.txtThirdPlaintext.setForeground(colorFG);
+				break;
+				
+			case 3:
+				cstb.txtFourthPlaintext.setBackground(colorBG);
+				cstb.txtFourthPlaintext.setForeground(colorFG);
+				break;
+			
+			default:
+				break;
+		}
+	}
+	
+	
+	
+	private boolean isOnePlaintextSelected(int idx) {
+		ArrayList<Boolean> clickedPlaintextsAtIdx = clickedPlaintexts.get(idx);
+		for(boolean b : clickedPlaintextsAtIdx) {
+			if(b == true)
+				return true;
+		}
+		
+		return false;
+	}
+	
+	
+	
+	
+	
+	private void txtPlaintextMouseDownAction(int idxToSet, CryptosystemTextbookComposite cstb, Color colorToSetBG, Color colorToResetBG, Color colorToSetFG, Color colorToResetFG) {
+		
+		int idx = cstb.cmbChooseBlock.getSelectionIndex();
+		if(isOnePlaintextSelected(idx) && !clickedPlaintexts.get(idx).get(idxToSet))
+			return;
+		
+		countClicksForPlaintexts[idxToSet] = (countClicksForPlaintexts[idxToSet] + 1) % 2;
+		
+		if(countClicksForPlaintexts[idxToSet] == 1) {
+			clickedPlaintexts.get(idx).set(idxToSet, true);
+			String key = String.valueOf(idx) + String.valueOf(idxToSet);
+			String val = plaintexts.get(idx).get(idxToSet);
+			currentSelectedPlaintexts.put(key, val);
+			String currentSelectedPlaintextsAsStr = getCurrentSelectedPlaintexts();
+			cstb.txtChosenPlaintexts.setText(currentSelectedPlaintextsAsStr);
+			setColorPlaintext(cstb, idxToSet, colorToSetBG, colorToSetFG);
+		}
+		
+		if(countClicksForPlaintexts[idxToSet] == 0) {
+			clickedPlaintexts.get(idx).set(idxToSet, false);
+			String key = String.valueOf(idx) + String.valueOf(idxToSet);
+			currentSelectedPlaintexts.remove(key);
+			String currentSelectedPlaintextsAsStr = getCurrentSelectedPlaintexts();
+			cstb.txtChosenPlaintexts.setText(currentSelectedPlaintextsAsStr);
+			setColorPlaintext(cstb, idxToSet, colorToResetBG, colorToResetFG);
+		}
+	}
+	
+	
+	
+	public void txtFirstPlaintextMouseDownAction(CryptosystemTextbookComposite cstb) {
+		if(clickedPlaintexts == null)
+			return;
+		
+		int idxToSet = 0;
+		txtPlaintextMouseDownAction(idxToSet, cstb, GUIHandler.colorSelectControlBG, GUIHandler.colorDeselectControlBG, GUIHandler.colorSelectControlFG, GUIHandler.colorDeselectControlFG);
```

Comment:
These lines are way too long. You can take the space for nicely formatted method calls to take some more lines, so that each parameter is on its own line. (parameter positioning mixup is a common cause for bugs)
```suggestion
		txtPlaintextMouseDownAction(
				idxToSet,
				cstb,
				GUIHandler.colorSelectControlBG,
				GUIHandler.colorDeselectControlBG,
				GUIHandler.colorSelectControlFG,
				GUIHandler.colorDeselectControlFG
		);
```